internal/pkg/cli/project_init.go

```diff
@@ -4,60 +4,166 @@
 package cli
 
 import (
+	"errors"
+	"fmt"
+
 	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store"
 	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store/ssm"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/color"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/log"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/prompt"
 	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/workspace"
 	"github.com/spf13/cobra"
 )
 
 // InitProjectOpts contains the fields to collect for creating a project.
 type InitProjectOpts struct {
 	ProjectName string
-	manager     archer.ProjectCreator
-	ws          archer.Workspace
+
+	projectStore archer.ProjectStore
+	ws           archer.Workspace
+	prompt       prompter
+}
+
+// NewInitProjectOpts returns a new InitProjectOpts.
+func NewInitProjectOpts() (*InitProjectOpts, error) {
+	ssmStore, err := ssm.NewStore()
+	if err != nil {
+		return nil, err
+	}
+
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, err
+	}
+	return &InitProjectOpts{
+		projectStore: ssmStore,
+		ws:           ws,
+		prompt:       prompt.New(),
+	}, nil
+}
+
+// Ask prompts the user for any required arguments that they didn't provide.
+func (opts *InitProjectOpts) Ask() error {
+	if opts.ProjectName != "" {
+		// User passed in a name as an argument.
+		return nil
+	}
+
+	// If there's a local project, we'll use that and just skip the project question.
+	// Otherwise, we'll load a list of existing projects that the customer can select from.
+	if summary, err := opts.ws.Summary(); err == nil {
```

Comment:
What does `err != nil` imply here? Seems like we just ignore that case?

Reply:
Correct, if there are errors while trying to read `./ecs-project/.ecs-workspace` then whatever 😜   
We will fetch the list of existing projects from SSM instead and prompt the user for the preference